core/modules/checkout/components/OrderReview.ts

```diff
@@ -37,7 +41,18 @@ export const OrderReview = {
         email: this.$store.state.checkout.personalDetails.emailAddress,
         password: this.$store.state.checkout.personalDetails.password,
         firstname: this.$store.state.checkout.personalDetails.firstName,
-        lastname: this.$store.state.checkout.personalDetails.lastName
+        lastname: this.$store.state.checkout.personalDetails.lastName,
+        addresses: [{
+          firstname: this.shippingDetails.firstName,
```

Comment:
this is nice method to refactor:
1. use mapped `getShippingDetails` vuex getter instead of `shippingDetails`
2. use mapped `getPersonalDetails` vuex getters instead of accessing state
3. ...and use mapped getters here when state is accessed directly
4. method is async, so we should use await for `this.$store.dispatch('user/register'` not use then after (with try/catch)